src/Aspire.Hosting/Dashboard/DashboardServiceData.cs

```diff
@@ -17,22 +16,70 @@ internal sealed class DashboardServiceData : IAsyncDisposable
 {
     private readonly CancellationTokenSource _cts = new();
     private readonly ResourcePublisher _resourcePublisher;
-    private readonly ConsoleLogPublisher _consoleLogPublisher;
+    private readonly ResourceLoggerService _resourceLoggerService;
 
     public DashboardServiceData(
-        DistributedApplicationModel applicationModel,
-        IKubernetesService kubernetesService,
         ResourceNotificationService resourceNotificationService,
         ResourceLoggerService resourceLoggerService,
-        IConfiguration configuration,
-        ILoggerFactory loggerFactory)
+        ILogger<DashboardServiceData> logger)
     {
-        var resourceMap = applicationModel.Resources.ToDictionary(resource => resource.Name, StringComparer.Ordinal);
-
+        _resourceLoggerService = resourceLoggerService;
         _resourcePublisher = new ResourcePublisher(_cts.Token);
-        _consoleLogPublisher = new ConsoleLogPublisher(_resourcePublisher, resourceLoggerService, kubernetesService, loggerFactory, configuration);
 
-        _ = new DcpDataSource(kubernetesService, resourceNotificationService, resourceMap, configuration, loggerFactory, _resourcePublisher.IntegrateAsync, _cts.Token);
+        var cancellationToken = _cts.Token;
+
+        Task.Run(async () =>
+        {
+            static GenericResourceSnapshot CreateResourceSnapshot(IResource resource, DateTime creationTimestamp, CustomResourceSnapshot snapshot)
+            {
+                ImmutableArray<EnvironmentVariableSnapshot> environmentVariables = [..
+                    snapshot.EnvironmentVariables.Select(e => new EnvironmentVariableSnapshot(e.Name, e.Value, e.IsFromSpec))];
+
+                ImmutableArray<ResourceServiceSnapshot> services =
+                [
+                    ..snapshot.Urls.Select(u => new ResourceServiceSnapshot(u.Name, u.Url, null)),
+                    ..snapshot.Services.Select(e => new ResourceServiceSnapshot(e.Name, e.AllocatedAddress, e.AllocatedPort))
+                ];
+
+                ImmutableArray<EndpointSnapshot> endpoints = [
+                    ..snapshot.Urls.Select(u => new EndpointSnapshot(u.Url, u.Url)),
+                    ..snapshot.Endpoints.Select(e => new EndpointSnapshot(e.EndpointUrl, e.ProxyUrl))
+                ];
+
+                var name = snapshot.Name ?? resource.Name;
+
+                return new GenericResourceSnapshot(snapshot)
+                {
+                    Uid = name,
+                    CreationTimeStamp = snapshot.CreationTimeStamp ?? creationTimestamp,
+                    Name = name,
+                    DisplayName = resource.GetReplicaCount() is 1 ? resource.Name : name,
+                    Endpoints = endpoints,
+                    Environment = environmentVariables,
+                    ExitCode = snapshot.ExitCode,
+                    ExpectedEndpointsCount = endpoints.Length,
+                    Services = services,
+                    State = snapshot.State ?? "Running"
```

Comment:
Should we really tell people that everything is A-OK if we do not know?

Reply:
I think not. Now we have this https://github.com/dotnet/aspire/commit/e86f9fd285fa32fc59d1151e9cfc92b9959ea9e4, so I'll remove the fallback.